spyderlib/spyder.py

```diff
@@ -2653,6 +2660,61 @@ def start_open_files_server(self):
                 self.sig_open_external_file.emit(fname)
             req.sendall(b' ')
 
+    # ---- Quit and restart
+    def restart(self):
+        """ """
+        # Get start path to use in restart script
+        spyder_start_directory = osp.dirname(osp.abspath(__file__))
+        spyder_folder = osp.split(spyder_start_directory)[0]
+        restart_script = osp.join(spyder_start_directory, 'spyder_restart.py')
+
+        # Get any initial argument passed when spyder was started
+        # Variables defined in bootstrap.py and spyderlib\start_app.py
+        env = os.environ.copy()
+        bootstrap_args = env.pop('SPYDER_BOOTSTRAP_ARGS', None)

```

Comment:
Thanks mind reader++, 

@ccordoba12 since I am passing the env to the new instance to be open, I want to clean it ...